Assets/Scripts/TorchManager.cs

```diff
@@ -6,14 +6,18 @@ public class TorchManager : MonoBehaviour
 {
     [SerializeField] private bool isTorchOn = false;
     [SerializeField] private Light TorchLight;
+    [SerializeField] private ParticleSystem fire_particles;
 
     private void Update()
     {
+        fire_particles = GetComponent<ParticleSystem>();
```

Comment:
you already assigned it in inspektor why you doining it there?
Also if you have to use `GetComponent` in update you made something wrong.